calicoctl/calicoctl/commands/node/diags.go

```diff
@@ -114,7 +114,7 @@ func runDiags(logDir string) error {
 		return fmt.Errorf("Error changing directory to temp directory to dump logs: %v", err)
 	}
 
-	err = os.Mkdir("diagnostics", os.ModeDir)
+	err = os.MkdirAll("diagnostics", os.ModeDir)
```

Comment:
I think that this works, but it is a little awkward since we seem to be only creating one level of directory at a time with `MkdirAll`. Maybe we should just call `MkdirAll("diagnostics/logs", os.ModeDir)` here instead so we don't have to call `MkdirAll` twice?

Reply:
Our goal is to prevent throwing that exception, not to create all the directories at once, and it's easier to read the codes, so I think it's better to use it twice, but it's up to you.